projects/aca-content/src/lib/components/header-actions/header-actions.component.html

```diff
@@ -0,0 +1,48 @@
+<div>
+  <div class="adf-toolbar--spacer adf-toolbar-divider"></div>
+  <span class="action-bar">
+    <button class="aca-mat-button aca-create-button" mat-stroked-button data-automation-id="create-button"
+      *ngIf=" isPersonalFilesRoute() || isFavoriteLibrariesRoute() || isLibrariesrRoute() "
+      [matMenuTriggerFor]="createMenu">
+      {{ 'APP.HEADER.BUTTONS.CREATE' | translate }}
+    </button>
+    <mat-menu #createMenu="matMenu" role="menu" class="app-create-menu__root-menu app-create-menu__sub-menu"
+      [overlapTrigger]="false" yPosition="below">
+      <div *ngFor="let action of createActions; trackBy: trackByActionId">
+        <app-toolbar-menu-item [actionRef]="action"></app-toolbar-menu-item>
+      </div>
+    </mat-menu>
+
+    <button class="aca-mat-button aca-upload-button" mat-stroked-button data-automation-id="upload-button"
+      *ngIf="isPersonalFilesRoute()" [matMenuTriggerFor]="uploadMenu">
+      {{ 'APP.HEADER.BUTTONS.UPLOAD' | translate }}
+    </button>
+    <mat-menu #uploadMenu="matMenu" role="menu" class="app-upload-menu__root-menu app-upload-menu__sub-menu"
+      [overlapTrigger]="false" yPosition="below">
+      <div *ngFor="let action of uploadActions; trackBy: trackByActionId">
+        <app-toolbar-menu-item [actionRef]="action"></app-toolbar-menu-item>
+      </div>
+    </mat-menu>
+
+    <ng-container *ngIf="(mainAction$ | async) as action">
+      <ng-container [ngSwitch]="action.type">
+          <ng-container *ngSwitchCase="actionTypes.button">
+              <button *ngIf="isTasksRoute() || isProcessesRoute()"
+                  mat-stroked-button
+                  [id]="action.id"
+                  (click)="runAction(action.actions.click)"
+                  [disabled]="action.disabled"
+                  class="app-main-action-button aca-mat-button aca-process-button"
+                  data-automation-id="app-main-action-button"
+              >
+                  {{action.title | translate}}
+              </button>
+          </ng-container>
+      </ng-container>
+    </ng-container>
+
+    <adf-toolbar-divider *ngIf=" isPersonalFilesRoute() || isFavoriteLibrariesRoute() || isLibrariesrRoute() || isTasksRoute() || isProcessesRoute() ">
```

Comment:
task and process should not be here

Reply:
please introduce a single function "canShowSearchSeparator(): boolean" and evaluate all conditions there